README.md

```diff
@@ -0,0 +1,80 @@
+# terraform-aws-ldap-maintenance
+
+A step function to maintain LDAP users via slack.
+
+## Overview
+
+This project deploys a collection of lambda functions, an api endpoint, and a step function that will automate disabling LDAP users via an interactive slack message.
+
+## Architecture
+
+![State Machine Definition](_docs/state_machine_def_0.0.1.png)
+
+## Submodules
+
+- [API Gateway](/modules/api_gateway): Deploys an API endpoint that responds asynchronously to slack events and triggers the Slack Listener lambda function
+
+### Lambda Functions
+
+- [DynamoDB Cleanup](/modules/lambda_functions/dynamodb_cleanup): Facilitates removing disabled users' email(s) from a target DynamoDB table
+- [LDAP Query](/modules/lambda_functions/ldap_query): Used to perform actions against a target ldap database
+- [Slack Listener](/modules/lambda_functions/slack_listener): Responds to slack events via an asynchronously executed lambda function
+- [Slack Notifier](/modules/lambda_functions/slack_notifier): Sends status updates to slack and a target step function
+
+## Setup
+
+1. Retrieve the LDAPS endpoint of your target AD deployment.
+
+    **Note:** This can be accomplished via SimpleAD by creating an ALB that listens via TLS on port 636 and forwards requests to your SimpleAD A record. See the associated [AWS blog post](https://aws.amazon.com/blogs/security/how-to-configure-an-ldaps-endpoint-for-simple-ad/) or the tests of this project for a reference architecture.
+
+2. Within your LDAP directory create a user that will be used by the lambda function. This user will need permissions to query LDAP and disable users.
```

Comment:
Are these specific permissions? If so, can we add them to the documentation?

Reply:
added a blurb about permission delegation that should cover it